data-otservbr-global/scripts/creaturescripts/monster/urmahlullu_change.lua

```diff
@@ -1,109 +1,77 @@
-                                          secondaryDamage, secondaryType)
-	if creature and creature:getName() == 'Urmahlullu the Immaculate' then
-		if creature:getHealth() <= healthStages[2] then
-			position = creature:getPosition()
-			creature:remove()
-			Game.createMonster('Wildness of Urmahlullu', position, true, true)
-			time = os.time()
-			-- make change and start counter (~ 1 minute)
-		end
-	end
-	if creature and creature:getName() == 'Wildness of Urmahlullu' then
-		if creature:getHealth() <= healthStages[3] then
-			if os.time() <= time + timetochange  then
-				position = creature:getPosition()
-				creature:remove()
-				Game.createMonster('Urmahlullu the Tamed', position, true, true)
-				time = os.time()
-				-- make change to urmahlullu the tamed and start new count (~1 minute)
-			else
-				position = creature:getPosition()
-				creature:remove()
-				Game.createMonster('Urmahlullu the Immaculate', position, true, true)
-				-- back to wildness of urmahlullu
-			end
-		end
-	end
-	if creature and creature:getName() == 'Urmahlullu the Tamed' then
-		if creature:getHealth() <= healthStages[4] then
-			if os.time() <= time + timetochange then
-				position = creature:getPosition()
-				creature:remove()
-				Game.createMonster('Wisdom of Urmahlullu', position, true, true)
-				time = os.time()
-				-- make change to wisdom of urmahlullu and start new count (~1 minute)
-			else
-				position = creature:getPosition()
-				creature:remove()
-				Game.createMonster('Wildness of Urmahlullu', position, true, true)
-				time = os.time()
-				-- back to wildness of urmahlullu
-			end
-		end
+local stages = {
+	{ name = "Urmahlullu the Immaculate", health = 515000 * healthMultiplier },
+	{ name = "Wildness of Urmahlullu", health = 400000 * healthMultiplier },
+	{ name = "Urmahlullu the Tamed", health = 300000 * healthMultiplier },
+	{ name = "Wisdom of Urmahlullu", health = 200000 * healthMultiplier },
+	{ name = "Urmahlullu the Weakened", health = 100000 * healthMultiplier },
+}
+
+local changeEvent = nil
+local revertEvent = nil
+
+local function revert(cid, name)
+	local creature = Creature(cid, name)
+	if not creature then return end
+
+	if creature:getName() == name then
+		return
 	end
-	if creature and creature:getName() == 'Wisdom of Urmahlullu' then
-		if creature:getHealth() <= healthStages[5] then
-			if os.time() <= time + timetochange then
-				position = creature:getPosition()
-				creature:remove()
-				Game.createMonster('Urmahlullu the Weakened', position, true, true)
-				time = os.time()
-				-- make change to urmahlullu the weakened and start new count (~1 minute)
-			else
-				position = creature:getPosition()
-				creature:remove()
-				Game.createMonster('Urmahlullu the Tamed', position, true, true)
-				time = os.time()
-				--back to urmahlullu the tamed
-			end
-		end
+
+	local position = creature:getPosition()
+	creature:remove()
+	Game.createMonster(name, position, true, true)
```

Comment:
Although it is not mandatory, it can be interesting to check the return of createMonster, since a pointer to Monster* is returned here.

```suggestion
	local monster = Game.createMonster(name, position, true, true)
	if not monster then return end
```